### ISSUE TITLE
nn_tutorial.py: better explanation of introduction of model.train

### ISSUE THREAD

**Issue by rodrigodesalvobraz** [pytorch/tutorials#524]
The nn_tutorial says, when validation is introduced:

"Note that we always call model.train() before training"

This may be confusing to learners because there was already training going on before the validation section but model.train() was not being used, so the "always" in the sentence is not quite correct. Besides, the learner may wonder why we were getting away with not using model.train() before, and why it is necessary now.

It seems that model.train() was added because of the use of model.eval(), so it would be good to mention that as a reason to add it at this point.

**Comment by jlin27** [comment]
Thanks for the suggestion @rodrigodesalvobraz ! We are reviewing how best to clarify this text. Will update once we come to a conclusion.

**Note:** From "Add validation" section of this tutorial: https://pytorch.org/tutorials/beginner/nn_tutorial.html#add-validation

**Consider clarifying w/ this linked post:**
Note that we always call model.train() before training, and model.eval() before inference, because these are used by layers such as nn.BatchNorm2d and nn.Dropout to ensure appropriate behaviour for these different phases. Also note that for training, model.train() is the same as model(). [Further explanation here](https://discuss.pytorch.org/t/model-train-and-model-eval-vs-model-and-model-eval/5744).

**Comment by holly1238** [comment]
Current tutorial has has these updates included
